### PR TITLE
Disable PCH if an extension is not enabled via options

### PR DESCRIPTION
**Pull request by AlexeySotkin**
Signed-off-by: Alexey Sotkin <alexey.sotkin@intel.com>

### CODE REVIEW

File: options_compile.cpp [hunk 1]
```diff
@@ -227,34 +227,73 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
...
+    "cl_intel_planar_yuv",
+    "cl_intel_subgroups",
+    "cl_intel_subgroups_short"
+  };
+  bool useModules = true;
+  auto cl_ext_it = std::find_if(effectiveArgs.begin(), effectiveArgs.end(),
+                   [](const ArgsVector::value_type& a) {
+                     return a.find("-cl-ext=") == 0 && a != "-cl-ext=-all";
+                   });
+  if (cl_ext_it != effectiveArgs.end()) {
```
[the remark this answers is not in the document]
Comment (woshiyifei) [reply options_compile.cpp:255]: In line 210, we'll add -cl-ext=-all to effectiveArgs, it'll exist as a single element. So there will be "[19] = "-cl-ext=-all","  and "[29] = "-cl-ext=-all,+cl_khr_byte_addressable_store,+cl_khr_fp16,....." in effectiveArgs.

[the remark this answers is not in the document]
Comment (AlexeySotkin) [reply options_compile.cpp]: This algorithm has been rewritten.

File: options_compile.cpp [hunk 2]
```diff
@@ -227,34 +229,86 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
...
+    llvm::StringRef clExtRef(clExtStr);
+    clExtRef.consume_front("-cl-ext=");
+    llvm::SmallVector<llvm::StringRef, 32> parsedExt;
+    clExtRef.split(parsedExt, ',');
+    for (llvm::StringRef ext : parsedExt) {
+      char sign = ext.front();
+      bool enabled = sign != '-';
+      llvm::StringRef extName = ext;
+      if (sign == '+' || sign == '-')
+        extName = extName.drop_front();
```
Comment (AlexeySachkov) [inline options_compile.cpp:262]:
```suggestion
      extName = extName.drop_front();
```

I guess we expect particular format here and `+` or `-` is required. I suggest either remove `if` or replace it with an `assert`

Comment (AlexeySotkin) [reply options_compile.cpp:262]: AFAIK, for Clang's `-cl-ext` option, missing `+` or `-` is legal and equivalent to `+`. I'm following the same logic here.

**Pull request merged**